tailcfg/tailcfg.go

```diff
@@ -574,9 +629,25 @@ func (req *RegisterRequest) Clone() *RegisterRequest {
 		tok := *res.Auth.Oauth2Token
 		res.Auth.Oauth2Token = &tok
 	}
+	res.DeviceCert = append(res.DeviceCert[:0:0], res.DeviceCert...)
+	res.Signature = append(res.Signature[:0:0], res.Signature...)
 	return res
 }
 
+// GenerateHash generates the hash required sign or verify a RegisterRequest
+// with SignatureV1.
+func (req *RegisterRequest) GenerateHash(
+	h hash.Hash, serverURL string, serverPubKey, machinePubKey wgkey.Key) ([]byte, error) {
+	h.Reset()
+
+	_, err := fmt.Fprint(h, req.Timestamp.UTC(), serverURL, req.DeviceCert, serverPubKey, machinePubKey)
```

Comment:
fmt.Fprint is going to print it in a very Go-specific way without specific formats.

If we ever want to interoperate with a different language (probably not?), then we'd want to do this in some predictable format with a fmt.Fprintf pattern specified. Maybe at least a TODO or comment that we're aware of this?

Reply:
Switched it to fmt.Fprintf and used a standard timestamp format